App.js

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import logo from './logo.svg';
+import './App.css';
+
+function App() {
+  return (
+    /*<div className="App">
+      <header className="App-header">
+        <img src={logo} className="App-logo" alt="logo" />
+        <p>
+          Edit <code>src/App.js</code> and save to reload.
+        </p>
+        <a
+          className="App-link"
+          href="https://reactjs.org"
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          Learn React
+        </a>
+      </header>
+    </div>*/
+    <div>
+      <PageTitle />
+      <PageTasks /> 
+    </div>
+  );
+}
+
+export default App;
+
+//class page title
+class PageTitle extends React.Component{
+  render(){
+    return (
+      <div className="PageTitle-div">
+        <h2>To Do List</h2>
+        <p>{new Date().getFullYear() + "-" + (new Date().getMonth() + 1 ) + "-" + new Date().getDate()}</p>
+      </div>
+    )
+  }
+}
+
+//add a new task
+class AddNewTask extends React.Component{
+  constructor(){
+    super();
+    this.state={
+      text: ""
+    }
+    this.InputChange = this.InputChange.bind(this);
+    this.SubmitNewTask = this.SubmitNewTask.bind(this);
+  }
+
+  InputChange(event){
+    this.setState({text: event.target.value});
+  }
+
+  SubmitNewTask(event){
+    this.props.GetNewTask(this.state.text);
+    this.setState({text: ""});
+  }
+
+
+  render(){
+    return (
+      <div className="AddNewTask-box">
+        <span className="AddNewTask-items">Add a new Task:</span>
+        <input placeholder="Type something about your new task..." className="AddNewTask-items" onChange={this.InputChange} value={this.state.text}/> 
+        <button className="AddNewTask-items" onClick={this.SubmitNewTask}>Add</button>
+      </div>
+      )
+  }
+}
+
+//task div
+class PageTasks extends React.Component{
+  constructor(){
+    super();
+    this.state={
+      newtasks:[],
+      donetasks:[]
+    }
+
+    this.AddNewTaskItem = this.AddNewTaskItem.bind(this);
+    this.SetTaskAsDone = this.SetTaskAsDone.bind(this);
+  }
+
+  AddNewTaskItem(text){
+    var addToTaskList = true;
+    this.state.newtasks.forEach(task => {
+      if(task.Content == text){
+        addToTaskList =  false;
+        return;
+      }
+    });
+    if(addToTaskList){
+      this.setState({newtasks: this.state.newtasks.concat({Content: text})});
+    }else{
+      alert("This task already exists!");
+    }
+    
+  }
+
+  SetTaskAsDone(text){
+    //update newTask list
+    var updateNewTaskList = [];
+    this.state.newtasks.forEach(task => {
+      if(task.Content != text){
+        updateNewTaskList.push(task);
+      }
+    });
+    this.setState({newtasks: updateNewTaskList});
+    //update doneTask list
+    this.setState({donetasks: this.state.donetasks.concat({Content: text})});
```

Comment:
I would write this part this way using spread operator https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax
```
this.setState({
  donetasks: [...this.state.donetasks, { Content: text }]
})
```

`.concat` surely works but this is a better and clearer way of writing more modern and idiomatic javascript